DVR/Interaction.swift

```diff
@@ -81,11 +81,18 @@ extension Interaction {
             "recorded_at": recordedAt.timeIntervalSince1970
         ]
 
-        var response = self.response.dictionary
-        if let data = responseData, let body = Interaction.encodeBody(data, headers: response["headers"] as? [String: String]) {
-            response["body"] = body
+        var responseDictionary = self.response.dictionary
+
+        if let httpResponse = response as? HTTPURLResponse {
+            responseDictionary["headers"] = httpResponse.allHeaderFields
+            responseDictionary["status"] = httpResponse.statusCode
+        }
+
+        if let data = responseData, let body = Interaction.encodeBody(data, headers: responseDictionary["headers"] as? [String: String]) {
+            responseDictionary["body"] = body
         }
-        dictionary["response"] = response
+
+        dictionary["response"] = responseDictionary
```

Comment:
@soffes Can you double check me on this? We can't have an `override var dictionary: [String: Any]` on `Foundation.HTTPURLResponse` because it is already inheriting a property of the same name/type from an extension on `Foundation.URLResponse` and you can't override properties/methods in extensions. 

I just wanna make sure this is the only spot we expected that override to be invoked and that this achieves the same thing.